### PR TITLE
Fix/db delete logs by batch ID

### PR DESCRIPTION
**Pull request by clpolet**
DB storage can now delete logs per batch Id.

### CODE REVIEW

File: MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m [hunk 1]
```diff
@@ -1,118 +1,182 @@
...
-@synthesize connection;
-
 #pragma mark - Initialization
 
 - (instancetype)init {
   self = [super init];
   if (self) {
-    self.connection = [[MSSqliteConnection alloc] initWithDatabaseFilename:kMSDBFileName];
+    _connection = [[MSSqliteConnection alloc] initWithDatabaseFilename:kMSDBFileName];
+    _batches = [NSMutableDictionary<NSString *, NSArray<NSString *> *> new];
```
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m]: Why does the storage need to pay attention to batch things? This is a log manager role.

File: MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m [hunk 2]
```diff
@@ -1,118 +1,182 @@
...
+      [[self getLogsFromDBWithGroupId:groupId limit:(limit + 1)] mutableCopy];
+  BOOL logsAvailable = NO;
   BOOL moreLogsAvailable = NO;
+  NSString *batchId;
 
-  // Remove the log in excess, it means there is more logs available.
-  if (logs.count > limit) {
-    [logs removeLastObject];
+  // More logs available for the next batch, remove the log in excess for this batch.
+  if (logs.count > 0 && logs.count > limit) {
```
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m]: I know this is not related to your change/fix but why does the storage class concern about next batch?

File: MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m (hunk 1 above)
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m]: That's not new, it's the design in place. We can revisit this one off line think.

File: MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m (hunk 2 above)
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m]: It's an optimization, this info is used by the consumer of the logs, if it knows that there is no next batch then no need  to trigger any timer to handle the next batch.

File: MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m [hunk 3]
```diff
@@ -1,118 +1,182 @@
...
+}
+
+#pragma mark - DB selection
+
+- (NSDictionary<NSString *, id<MSLog>> *)getLogsFromDBWithGroupId:(NSString *)groupId {
   NSString *selectLogQuery =
-      [NSString stringWithFormat:@"select * from %@ where %@ == '%@'", kMSLogTableName, kMSGroupIdColumnName, groupId];
-  return [self getLogsWithQwery:selectLogQuery];
+      [NSString stringWithFormat:@"SELECT * FROM %@ WHERE %@ == '%@'", kMSLogTableName, kMSGroupIdColumnName, groupId];
+  return [self getLogsFromDBWithQwery:selectLogQuery];
```
Comment (clpolet) [inline MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m]: Good catch :)

File: MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m (hunk 1 above)
Comment (jaelim-ms) [inline MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m]: Discussed offline. We will take care of batch ID thing in a separate task.

**Pull request merged**